torch_xla/distributed/xla_backend.py

```diff
@@ -0,0 +1,289 @@
+import distutils.util
+import os
+import torch
+import torch.distributed as dist
+import torch_xla
+import torch_xla.core.xla_model as xm
+import logging
+from torch._C._distributed_c10d import (
+    ProcessGroup,
+    Work,
+)
+
+
+def _create_xla_process_group(prefix_store, rank, size, timeout):
+  return ProcessGroupXla(prefix_store, rank, size, timeout)
+
+
+def _register_xla_backend():
+  dist.Backend.register_backend('xla', _create_xla_process_group)
+
+
+_register_xla_backend()
+
+
+class ProcessGroupXla(ProcessGroup):
+  '''ProcessGroup for XLA devices. See ProcessGroup for doc.
+
+    https://github.com/pytorch/pytorch/blob/074c77601114c66df346c5465667ce8cfaa6d22c/torch/csrc/distributed/c10d/ProcessGroup.hpp#L79
```

Comment:
We have new tutorial that we can link: https://pytorch.org/tutorials//intermediate/process_group_cpp_extension_tutorial.html